lemarche/api/siaes/views.py

```diff
@@ -38,64 +49,110 @@ def _wrapper(*args, **kwargs):
     return _wrapper
 
 
-#  FIXME:Given the peculiar user-token access, it's not possible
-# to use generic class-based views at this moment.
-class SiaeList(APIView):
+class Siae(mixins.ListModelMixin, mixins.RetrieveModelMixin, viewsets.GenericViewSet):
     """
-    Lister les SIAES
+    Données d'une structure d'insertion par l'activité économique (SIAE).
     """
 
-    def get(self, request, format=None):
+    queryset = Directory.objects.all()
+    serializer_class = SiaeListSerializer
+    filter_class = SiaeFilter
+
+    def get_object_by_id(self, pk):
+        try:
+            return Directory.objects.get(pk=pk)
+        except Directory.DoesNotExist:
+            raise Http404
+
+    def get_object_by_siret(self, siret):
+        try:
+            return Directory.objects.get(siret=siret)
+        except Directory.DoesNotExist:
+            raise Http404
+
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(name="token", description="Token Utilisateur", required=False, type=str),
+        ]
+    )
+    def list(self, request, format=None):
+        """
+        Liste exhaustive des structures d'insertion par l'activité économique (SIAE).
+        """
         if request.method == "GET":
-            siaes = Directory.objects.all()
+            queryset = self.filter_queryset(self.get_queryset())
```

Comment:
Il est peut-être possible de raccourcir un peu le code ainsi : 

```suggestion
            token = request.GET.get("token", None)
            queryset = self.filter_queryset(self.get_queryset())
            if not token:
                queryset = queryset[:10]
```

A tester. :)